src/domains/citizen-auth-app/10_function_public.tf

```diff
@@ -0,0 +1,258 @@
+#
+# APP CONFIGURATION
+#
+
+locals {
+  function_public = {
+    app_settings_common = {
+      FUNCTIONS_WORKER_RUNTIME       = "node"
+      FUNCTIONS_WORKER_PROCESS_COUNT = 4
+      NODE_ENV                       = "production"
+
+      // Keepalive fields are all optionals
+      FETCH_KEEPALIVE_ENABLED             = "true"
+      FETCH_KEEPALIVE_SOCKET_ACTIVE_TTL   = "110000"
+      FETCH_KEEPALIVE_MAX_SOCKETS         = "40"
+      FETCH_KEEPALIVE_MAX_FREE_SOCKETS    = "10"
+      FETCH_KEEPALIVE_FREE_SOCKET_TIMEOUT = "30000"
+      FETCH_KEEPALIVE_TIMEOUT             = "60000"
+
+      # UNIQUE EMAIL ENFORCEMENT
+      PROFILE_EMAIL_STORAGE_CONNECTION_STRING = data.azurerm_storage_account.citizen_auth_common.primary_connection_string
+      PROFILE_EMAIL_STORAGE_TABLE_NAME        = "profileEmails"
+
+      COSMOSDB_URI      = data.azurerm_cosmosdb_account.cosmos_api.endpoint
+      COSMOSDB_KEY      = data.azurerm_cosmosdb_account.cosmos_api.primary_key
+      COSMOSDB_NAME     = "db"
+      StorageConnection = data.azurerm_storage_account.storage_api.primary_connection_string
+
+      VALIDATION_CALLBACK_URL = "https://api-app.io.pagopa.it/email_verification.html"
+      CONFIRM_CHOICE_PAGE_URL = "https://api-app.io.pagopa.it/email_confirm.html"
+    }
+  }
+}
+
+resource "azurerm_resource_group" "public_rg_itn" {
+  name     = format("%s-public-rg-01", local.short_project_itn)
+  location = local.itn_location
+
+  tags = var.tags
+}
+
+
+module "function_public_snet_itn" {
+  source                                    = "git::https://github.com/pagopa/terraform-azurerm-v3.git//subnet?ref=v8.44.0"
+  name                                      = format("%s-public-snet-01", local.short_project_itn)
+  address_prefixes                          = var.cidr_subnet_fnpublic_itn
+  resource_group_name                       = data.azurerm_virtual_network.common_vnet_italy_north.resource_group_name
+  virtual_network_name                      = data.azurerm_virtual_network.common_vnet_italy_north.name
+  private_endpoint_network_policies_enabled = true
+
+  service_endpoints = [
```

Comment:
What are the downstream dependencies of this function app? We can remove Cosmos DB for sure, maybe the others as well

Reply:
cosmos is used to create a profile model and retrieve some data. storage is used to retrieve data from a table storage. I believe we need to keep them

Reply:
If downstream resources use private endpoints, we don't need to set the service endpoints (all Cosmos instances use peps as far as I know). Please check if the storage account provides a private endpoint for the table endpoint